packages/block-library/src/column/edit.native.js

```diff
@@ -0,0 +1,144 @@
+
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect } from '@wordpress/data';
+import { compose, withPreferredColorScheme } from '@wordpress/compose';
+import { InnerBlocks } from '@wordpress/block-editor';
+import { withViewportMatch } from '@wordpress/viewport';
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+
+function ColumnEdit( {
+	hasInnerBlocks,
+	isSelected,
+	getStylesFromColorScheme,
+	isParentSelected,
+	isDescendantOfParentSelected,
+	columnsCount,
+	columnsContainerWidth,
+	isMobile,
+} ) {
+	const columnContainerBaseWidth = styles[ 'column-container-base' ].maxWidth;
+	const containerMaxWidth = styles[ 'columns-container' ].maxWidth;
+
+	const containerWidth = columnsContainerWidth || containerMaxWidth;
+
+	const minWidth = Math.min( containerWidth, columnContainerBaseWidth );
+
+	const getColumnsInRow = ( columnsNumber ) => {
+		if ( minWidth < 480 ) {
+			return 1;
+		}
+		if ( minWidth >= 480 && minWidth < 768 ) {
+			return 2;
+		}
+		return columnsNumber;
+	};
```

Comment:
`getColumnsInRow` and the cut off points needs to be implement in more flexible way - we are working on that with Luke